language/move-binary-format/src/compatibility.rs

```diff
@@ -174,27 +178,47 @@ impl Compatibility {
                     &new_func.type_parameters,
                 )
             {
-                struct_and_function_linking = false;
+                if matches!(old_func.visibility, Visibility::Friend) {
+                    friend_linking = false;
+                } else {
+                    struct_and_function_linking = false;
+                }
             }
         }
 
-        if check_friend_linking {
-            // check friend declarations compatibility
-            //
-            // - additions to the list are allowed
-            // - removals are not allowed
-            //
-            let old_friend_module_ids: BTreeSet<_> = old_module.friends.iter().cloned().collect();
-            let new_friend_module_ids: BTreeSet<_> = new_module.friends.iter().cloned().collect();
-            if !old_friend_module_ids.is_subset(&new_friend_module_ids) {
-                struct_and_function_linking = false;
-            }
+        // check friend declarations compatibility
+        //
+        // - additions to the list are allowed
+        // - removals are not allowed
+        //
+        let old_friend_module_ids: BTreeSet<_> = old_module.friends.iter().cloned().collect();
+        let new_friend_module_ids: BTreeSet<_> = new_module.friends.iter().cloned().collect();
+        if !old_friend_module_ids.is_subset(&new_friend_module_ids) {
+            friend_linking = false;
+        }
+
+        //if struct_and_function_linking = false, friend_linking makes no sense.
```

Comment:
I'm not sure I fully understand the need for this--is there a test that exercises this path?

Reply:
Now it is not needed, the previous version used this to avoid return

Compatibility{struct_and_function_linking:false, friend_linking:true }.

Reply:
Can we delete this if it is no longer needed?

Reply:
Done